.github/workflows/causal_lm_cpp.yml

```diff
@@ -274,6 +274,41 @@ jobs:
           && call .\ov\setupvars.bat
           && python samples\python\greedy_causal_lm\lora.py .\TinyLlama\TinyLlama-1.1B-intermediate-step-1431k-3T\ adapter_model.safetensors "How to create a table with two columns, one of them has type float, another one has type int?"
 
+  cpp-greedy_causal_lm-Chatglm3-6b:
+    runs-on: ubuntu-20.04-16-cores
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - name: Install OpenVINO
+        run: |
+          mkdir ./ov/
+          curl ${{ env.l_ov_link }} | tar --directory ./ov/ --strip-components 1 -xz
+          sudo ./ov/install_dependencies/install_openvino_dependencies.sh
+      - name: Build app
+        run: |
+          source ./ov/setupvars.sh
+          cmake -DCMAKE_BUILD_TYPE=Release -S ./ -B ./build/
+          cmake --build ./build/ --config Release -j
+      - name: Download and convert and model
+        run: |
+          source ./ov/setupvars.sh
+          python -m pip install --upgrade-strategy eager -r ./samples/requirements.txt
+          python -m pip install ./thirdparty/openvino_tokenizers/[transformers] --pre --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly
+          optimum-cli export openvino --trust-remote-code --weight-format fp16 --model THUDM/chatglm3-6b chatglm3-6b --task text-generation-with-past
+      - run: > 
+          . ./ov/setupvars.sh
+          && timeout 2m ./build/samples/cpp/greedy_causal_lm/greedy_causal_lm ./chatglm3-6b/ 69 | diff <(timeout 2m samples/python/greedy_causal_lm/greedy_causal_lm.py ./chatglm3-6b/ 69) -
```

Comment:
> @Wovchena can you please guide me that what possibly gone wrong for `cpp-greedy_causal_lm-Chatglm3-6b` and `cpp-prompt_lookup_decoding_lm-ubuntu` although both **greedy_causal_lm** and **prompt_lookup_decoding_lm** works for chatglm3-6b in my local machine

The samples work but the comparison fails which means the samples produced different results. Try running the samples again, but pay attention to their output.

By the way, here's a newer and more verbose implementation of the same idea using `tee` and `timeout-minutes`. It is also split into multiple steps to make it clear which command fails. You can rewrite your tests in a similar way to simplify reasoning about what happened. https://github.com/openvinotoolkit/openvino.genai/blob/6165c479f2d99b91712bc105fe7eba9139952487/.github/workflows/causal_lm_cpp.yml#L776-L790